### PR TITLE
Adding extra_hosts param to docker-compose

### PR DESCRIPTION
**Pull request by birdy-**
Adding extra_hosts parameter with doc and tests

### CODE REVIEW

File: compose/service.py [hunk 1]
```diff
@@ -40,6 +40,7 @@
     'volumes',
     'volumes_from',
     'working_dir',
+    'extra_hosts',
```
Comment (dnephin) [inline compose/service.py]: Please keep this list sorted alphabetically (same with the list below)

**Comment by dnephin** [comment]
rebase with master to pickup the fix for those test failures, otherwise looks good

**Comment by birdy-** [comment]
Rebase done

**Comment by birdy-** [comment]
Fixed alphabetical order too

**Comment by aanand** [comment]
Test appears to be [failing](https://app.wercker.com/#build/54dcd3b23e143292230be65b):

```
FAIL: test_create_container_with_extra_hosts (tests.integration.service_test.ServiceTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/code/tests/integration/service_test.py", line 144, in test_create_container_with_extra_hosts
    ['myserver:192.168.2.90', 'myserver2:10.55.32.12'])
AssertionError: None != [u'myserver:192.168.2.90', u'myserver2:10.55.32.12']
```

**Comment by dnephin** [comment]
I think this feature requires docker 1.3 (http://docs.docker.com/v1.3/release-notes/), and if I remember correctly wercker is running docker 1.2, so I think this is blocking on #886 as well

**Comment by birdy-** [comment]
Closed since this feature exists now in docker-compose.

**Pull request closed without merging**